library/alloc/src/iter.rs

```diff
@@ -0,0 +1,40 @@
+//! Allocation extensions for [`Iterator`].
+//!
+//! *[See also the Iterator trait][Iterator].*
+#![unstable(feature = "iterator_join", issue = "75638")]
+
+use crate::slice::Join;
+use crate::vec::Vec;
+
+/// Iterator extension traits that requires allocation.
+#[unstable(feature = "iterator_join", issue = "75638")]
+pub trait IteratorExt: Iterator {
```

Comment:
```suggestion
pub trait IteratorAlloc: Iterator {
```
I'd caution against calling this trait `IteratorExt`. The name isn't wrong—this *is* an extension trait of `Iterator` for contexts where `alloc` is available—but it's not the *only* reasonable extension trait of `Iterator` that could exist: what about an extension trait for `Iterator` that's usable in contexts where `std` is also available?